PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/springcloud/dependency/action/AddAzureDependencyAction.java

```diff
@@ -71,81 +70,72 @@
     public boolean onActionPerformed(@NotNull AnActionEvent event, @Nullable Operation operation) {
         final Module module = event.getData(LangDataKeys.MODULE);
         final Project project = module.getProject();
-        final MavenProjectsManager projectsManager = MavenProjectsManager.getInstance(module.getProject());
+        final MavenProjectsManager projectsManager = MavenProjectsManager.getInstance(project);
         final MavenProject mavenProject = projectsManager.findProject(module);
-        assert mavenProject != null;
-
-        MavenEmbeddersManager embeddersManager = projectsManager.getEmbeddersManager();
-        MavenExplicitProfiles profiles = mavenProject.getActivatedProfilesIds();
-        MavenProjectsProcessor myFoldersResolvingProcessor = new MavenProjectsProcessor(project, "Generating Maven sources...", true, embeddersManager);
-        MavenEmbedderWrapper embedder = embeddersManager.getEmbedder(mavenProject, MavenEmbeddersManager.FOR_DEPENDENCIES_RESOLVE);
-
+        if (mavenProject == null) {
+            PluginUtil.showErrorNotificationProject(project, "Error",
+                                             String.format("Project '%s' is not a maven project.", project.getName()));
+            return true;
+        }
         ProgressManager.getInstance().runProcessWithProgressSynchronously(() -> {
```

Comment:
`runProcessWithProgressSynchronously` seems will block users operation, shall we change it to background task?

Reply:
This task may finish in 10 seconds